src/plugins/gui/hooks/useFormFieldState.ts

```diff
@@ -0,0 +1,20 @@
+import { useState } from 'react'
+
+import { useHandler } from '../../../hooks/useHandler'
+
+interface FieldState {
+  onChangeText: (text: string) => void
+  value: string
+}
+export const useFormFieldState = (defaultValue: string = ''): FieldState => {
+  const [value, setValue] = useState<string>(defaultValue)
+
+  const handleChange = useHandler(text => {
+    setValue(text)
+  })
```

Comment:
I thought we're suppose to use `useHandler` for all event handler function by convention? The function reference is stable with `useHandler` so this abstraction leverages that stability.

I can remove this abstraction completely though.

Reply:
The callback produced by `useState` is itself a stable reference, so you aren't achieving anything by wrapping it in `useHandler`. This would only be valuable if we had other logic of our own, or needed to change the signature somehow.